code/modules/asset_cache/asset_list.dm

```diff
@@ -140,7 +161,26 @@ GLOBAL_LIST_EMPTY(asset_datums)
 	if (!should_refresh() && read_from_cache())
 		return
 
+	// If it's cached, may as well load it now, while the loading is cheap
+	if(CONFIG_GET(flag/cache_assets) && cross_round_cachable)
+		load_immediately = TRUE
+
 	create_spritesheets()
+	if(load_immediately)
+		realize_spritesheets(yield = FALSE)
+	else
+		SSasset_loading.generate_queue += src
+
+/datum/asset/spritesheet/proc/realize_spritesheets(yield)
+	if(fully_generated)
+		return
+	while(length(to_generate))
+		var/list/stored_args = to_generate[to_generate.len]
+		to_generate.len--
+		queuedInsert(arglist(stored_args))
+		// I await my deathsquad MSO
+		if(yield && MC_TICK_CHECK_FOR(SSasset_loading))
```

Comment:
This is technically unnecessary, normal tickcheck and checktick use the tick limit of the current subsystem if one is running.

If you want a if check instead of a sleep from the normal checktick define, you can just use the one that evaluates to true instead of the one that calls stoplag: https://github.com/tgstation/tgstation/blob/master/code/__DEFINES/_tick.dm#L18-L21

Reply:
I was concerned about not calling pause() there, it seemed like a potential footgun
if it's not I'll just replace it with TICK_CHECK, that was my initial instinct

Reply:
na, this is slightly cleaner then having to mc tick check again in the subsystem. we can keep it.

Reply:
although actually you do both it seems.

Reply:
need both, since I want to split up the generation of spritesheets into little bit